snuba/migrations/snuba_migrations/functions/0002_add_thread_id.py

```diff
@@ -0,0 +1,180 @@
+    Column("os_version", String(Modifiers(low_cardinality=True))),
+    Column("retention_days", UInt(16)),
+    Column("count", AggregateFunction("count", [Float(64)])),
+    Column(
+        "percentiles",
+        AggregateFunction("quantiles(0.5, 0.75, 0.9, 0.95, 0.99)", [Float(64)]),
+    ),
+    Column("min", AggregateFunction("min", [Float(64)])),
+    Column("max", AggregateFunction("max", [Float(64)])),
+    Column("avg", AggregateFunction("avg", [Float(64)])),
+    Column("sum", AggregateFunction("sum", [Float(64)])),
+    Column(
+        "worst", AggregateFunction("argMax", [Tuple((UUID(), UInt(64))), Float(64)])
+    ),
+    Column(
+        "examples", AggregateFunction("groupUniqArray(5)", [Tuple((UUID(), UInt(64)))])
+    ),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    index_granularity = "2048"
+    storage_set = StorageSetKey.FUNCTIONS
+
+    data_granularity = 60 * 60  # 1 hour buckets
+
+    local_raw_table = "functions_raw_local"
+    dist_raw_table = "functions_raw_dist"
+
+    local_materialized_table = "functions_mv_local"
+    dist_materialized_table = "functions_mv_dist"
+
+    local_view_table = "functions_local"
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=self.storage_set,
+                table_name=self.local_raw_table,
+                column=Column("thread_id", UInt(64)),
+                after="profile_id",
+            ),
+            operations.DropTable(
+                storage_set=self.storage_set, table_name=self.local_materialized_table
+            ),
+            operations.DropTable(
+                storage_set=self.storage_set, table_name=self.local_view_table
+            ),
+            operations.CreateTable(
+                storage_set=self.storage_set,
+                table_name=self.local_materialized_table,
+                columns=columns,
+                engine=table_engines.AggregatingMergeTree(
+                    storage_set=self.storage_set,
+                    order_by="(project_id, transaction_name, timestamp, depth, parent_fingerprint, fingerprint, name, package, path, is_application, platform, environment, release, os_name, os_version, retention_days)",
+                    primary_key="(project_id, transaction_name, timestamp)",
+                    partition_by="(retention_days, toMonday(timestamp))",
+                    settings={"index_granularity": self.index_granularity},
+                    ttl="timestamp + toIntervalDay(retention_days)",
+                ),
+            ),
+            operations.CreateMaterializedView(
+                storage_set=self.storage_set,
+                view_name=self.local_view_table,
+                destination_table_name=self.local_materialized_table,
+                columns=columns,
+                query=self.__MATVIEW_STATEMENT,
+            ),
```

Comment:
This is mostly identical to what it was previously. A few notable changes
- the worst + examples aggregations now return a tuple containing the profile id + thread id
- the primary key has been cut down to exclude some of the higher cardinality columns like fingerprint
- bumped the materialization version from 0 to 1 for the new data